app/assets/javascripts/models/ui/state_manager.coffee

```diff
@@ -177,7 +187,10 @@
       component = components.shift()
 
       state.visible = component != 'map'
-      state.parent = !component
+      if component == 'collections'
```

Comment:
Thinking about this, isn't this basically the only necessary trigger, with nothing necessary on the master overlay or in the template? If you came from /search/collections, the facets should be collapsed when you go back, otherwise they should be expanded. That also feels more correct / future-proof in case we use the parent for other things in the future.